torch/csrc/jit/codegen/cuda/glfdc/eval_vm.h

```diff
@@ -0,0 +1,734 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/glfdc/eval_uinstr.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+namespace glfdc {
+
+using MemoryBlock = DEVector<scalar_type>;
+
+template <typename BindingFn_>
+c10::optional<scalar_type> evaluateLoop(
+    const std::vector<UInst>& instructions,
+    const InitialStack<scalar_type>& init,
+    MemoryBlock& memory,
+    BindingFn_& fn,
+    EvalState& es) noexcept {
+  // Instruction pointer
+  unsigned ip = 0;
+
+  // Registers
+  scalar_type a, b, t;
+
+  // Stack of operands
+  OperandsStack operands(memory);
```

Comment:
Just curious if there's any particular reason for having 3 registers and 2 stacks. Is it because of more convenient register allocation or better support of control flows?